src/ppx/application.cpp

```diff
@@ -718,25 +714,25 @@ void Application::TakeScreenshot()
     bufferCi.memoryUsage                 = grfx::MEMORY_USAGE_GPU_TO_CPU;
     PPX_CHECKED_CALL(mDevice->CreateBuffer(&bufferCi, &screenshotBuf));
 
-    // We wait for idle so that we can avoid tracking swapchain fences.
-    // It's not ideal, but we won't be taking screenshots in
+    // We wait for idle so that we can avoid tracking image fences.
+    // It's not ideal, but we won't be taking saving images in
     // performance-critical scenarios.
     PPX_CHECKED_CALL(queue->WaitIdle());
 
-    // Copy the swapchain image into the buffer.
+    // Copy the image into the buffer.
     grfx::CommandBufferPtr cmdBuf;
     PPX_CHECKED_CALL(queue->CreateCommandBuffer(&cmdBuf, 0, 0));
 
     grfx::ImageToBufferOutputPitch outPitch;
     cmdBuf->Begin();
     {
-        cmdBuf->TransitionImageLayout(swapchainImg, PPX_ALL_SUBRESOURCES, grfx::RESOURCE_STATE_PRESENT, grfx::RESOURCE_STATE_COPY_SRC);
+        cmdBuf->TransitionImageLayout(image, PPX_ALL_SUBRESOURCES, grfx::RESOURCE_STATE_PRESENT, grfx::RESOURCE_STATE_COPY_SRC);
```

Comment:
Should this and the one on line 735 transition from/to resourceState parameter instead? Otherwise it is unused

Reply:
Good catch. Yes, you are right!